src/s1ap/handlers/erab_setup_response.c

```diff
@@ -0,0 +1,136 @@
+/*
+* Copyright 2020-present, Infosys Ltd.
+*
+* SPDX-License-Identifier: Apache-2.0
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
+
+#include "ipc_api.h"
+#include "s1ap.h"
+#include "sctp_conn.h"
+#include "s1ap_msg_codes.h"
+#include "msgType.h"
+
+extern ipc_handle ipc_S1ap_Hndl;
+
+void dump_erab_su_resp(struct erabSuResp_Q_msg *msg)
+{
+   log_msg(LOG_INFO, "MME-UE-S1AP-ID %d\n", msg->s1ap_mme_ue_id);
+   log_msg(LOG_INFO, "eNB-UE-S1AP-ID %d\n", msg->s1ap_enb_ue_id);
+   log_msg(LOG_INFO, "eRABSetupList_count %d\n", msg->erab_su_list.count);
+   for(int i = 0; i < msg->erab_su_list.count; i++)
+   {
+       log_msg(LOG_INFO,"eRAB_setup_item[%d].erab_id: %d\n", i, msg->erab_su_list.
+		       erab_su_item[i].e_RAB_ID);
+       log_msg(LOG_INFO,"eRAB_setup_item[%d].transportLayerAddress: %x\n", i, msg->erab_su_list.
+                       erab_su_item[i].transportLayerAddress);
+       log_msg(LOG_INFO,"eRAB_setup_item[%d].gtp_teid: %d\n", i, msg->erab_su_list.
+                       erab_su_item[i].gtp_teid);
+   }
+   for(int i = 0; i < msg->erab_fail_list.count; i++)
+   {
+       log_msg(LOG_INFO,"eRAB_failed_to_setup_item[%d].erab_id: %d\n", i, msg->erab_fail_list.
+		       erab_fail_item[i].e_RAB_ID);
+   }
+
+}
+
+int erab_setup_response_handler(SuccessfulOutcome_t *msg)
+{
+    s1_incoming_msg_data_t erab_su_resp = {0};
+    struct proto_IE erab_su_resp_ies = {0};
+
+    int decode_status = convertErabSetupRespToProtoIe(msg, &erab_su_resp_ies);
+    if (decode_status < 0)
+    {
+        log_msg(LOG_ERROR, "Failed to decode ERAB Setup Response\n");
+
+        if (erab_su_resp_ies.data != NULL)
```

Comment:
even if its null, you can just call free.

Reply:
Done.